sqlite3_opt_serialize_test.go

```diff
@@ -0,0 +1,120 @@
+// +build !libsqlite3 sqlite_serialize
+
+package sqlite3
+
+import (
+	"database/sql"
+	"database/sql/driver"
+	"os"
+	"testing"
+)
+
+func TestSerialize(t *testing.T) {
```

Comment:
Can we write this test to go through `database/sql` instead? That will also serve as a good example for developers for how to use it from their code.

Reply:
Sounds reasonable, let me take a look.

Reply:
@otoolep It doesn't look like this was done.

Reply:
Oh yes,  you're right -- I had forgotten this is on me. Thanks for reminding me.

Reply:
I rewrote the Serialization test. Do you want to take a look? Is this what you had in mind?

Reply:
New test added.